In [11]:
import pandas as pd
import os
from dotenv import load_dotenv
from sqlalchemy import create_engine,Table, Column, Integer, String, Float, MetaData


In [ ]:
#!pip show python-dotenv

# 1.- Accesos Por Tecnología

In [343]:
df_acceso_por_tecnologia=pd.read_excel('../DatasetCrudos/Internet.xlsx',sheet_name='Accesos Por Tecnología')

In [344]:
df_acceso_por_tecnologia.describe()

ADSL    Cablemodem  Fibra óptica      Wireless         Otros  \
count  9.600000e+02  9.600000e+02  9.600000e+02     960.00000    960.000000   
mean   1.171007e+05  1.816350e+05  4.428212e+04   12071.28750   6451.996875   
std    2.411026e+05  4.461057e+05  1.681546e+05   20544.30774  11192.518329   
min    3.296000e+03  0.000000e+00  0.000000e+00       0.00000      2.000000   
25%    2.123025e+04  1.054450e+04  3.470000e+02     676.00000    330.000000   
50%    4.547550e+04  3.967000e+04  2.541000e+03    4947.00000   1881.000000   
75%    9.190500e+04  7.549325e+04  2.347675e+04   13694.25000   7278.750000   
max    1.586343e+06  2.797700e+06  1.784593e+06  138405.00000  73415.000000   

              Total  
count  9.600000e+02  
mean   3.615411e+05  
std    7.778954e+05  
min    1.255700e+04  
25%    5.900025e+04  
50%    1.111385e+05  
75%    1.883518e+05  
max    5.011620e+06

In [345]:
df_acceso_por_tecnologia.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 962 entries, 0 to 961
Data columns (total 9 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Año           961 non-null    object 
 1   Trimestre     961 non-null    object 
 2   Provincia     960 non-null    object 
 3   ADSL          960 non-null    float64
 4   Cablemodem    960 non-null    float64
 5   Fibra óptica  960 non-null    float64
 6   Wireless      960 non-null    float64
 7   Otros         960 non-null    float64
 8   Total         960 non-null    float64
dtypes: float64(6), object(3)
memory usage: 67.8+ KB


En este punto podemos ver que hay 962 registros con 9 columnas , 3 son de tipo string y 6 son númericos

In [346]:
df_acceso_por_tecnologia.isna().sum()

Año             1
Trimestre       1
Provincia       2
ADSL            2
Cablemodem      2
Fibra óptica    2
Wireless        2
Otros           2
Total           2
dtype: int64

In [347]:
df_acceso_por_tecnologia['Año'].unique()

array([2023, 2022, 2021, 2020, 2019, '2019 *', 2018, 2017, 2016, 2015,
       2014, nan, '*'], dtype=object)

In [348]:
last_row=df_acceso_por_tecnologia.loc[df_acceso_por_tecnologia['Año']=='*']
print(last_row['Trimestre'])

961    Los datos provinciales no coinciden a nivel na...
Name: Trimestre, dtype: object


Borramos los registros con na ya que solo son 2 registros y con esto logramos tener datos desde 2014 hasta 2023

In [349]:
df_acceso_por_tecnologia=df_acceso_por_tecnologia.dropna()

In [350]:
df_acceso_por_tecnologia.loc[df_acceso_por_tecnologia['Año']=='2019 *']
#podemos ver que hay que limpiar Año y Trimestre

Año Trimestre            Provincia       ADSL  Cablemodem  \
408  2019 *       3 *         Buenos Aires  1277249.0   2144956.0   
409  2019 *       3 *      Capital Federal   300288.0   1080973.0   
410  2019 *       3 *            Catamarca    24349.0      5543.0   
411  2019 *       3 *                Chaco    50416.0     49156.0   
412  2019 *       3 *               Chubut    37298.0     27090.0   
..      ...       ...                  ...        ...         ...   
475  2019 *       1 *           Santa Cruz    10659.0      4615.0   
476  2019 *       1 *             Santa Fe   340038.0    340344.0   
477  2019 *       1 *  Santiago Del Estero    25337.0     61547.0   
478  2019 *       1 *     Tierra Del Fuego    19789.0     23549.0   
479  2019 *       1 *              Tucumán   159215.0     11670.0   

     Fibra óptica  Wireless    Otros      Total  
408      425386.0   85572.0  27070.0  3960233.0  
409       24900.0    3895.0  31155.0  1441211.0  
410          13.0     244.0   2251.0    32400.0  
411        8063.0    5224.0   1592.0   114451.0  
412        1917.0   15149.0   7551.0    89005.0  
..            ...       ...      ...        ...  
475         382.0    6514.0   2735.0    24905.0  
476       15621.0   23215.0  17860.0   737078.0  
477          23.0    2847.0     99.0    89853.0  
478        1623.0       0.0   1486.0    46447.0  
479         118.0     678.0    321.0   172002.0  

[72 rows x 9 columns]

In [351]:
df_acceso_por_tecnologia.loc[df_acceso_por_tecnologia['Año']=='2019 *','Año']=2019
df_acceso_por_tecnologia.loc[df_acceso_por_tecnologia['Año']=='2019 *']

Empty DataFrame
Columns: [Año, Trimestre, Provincia, ADSL, Cablemodem, Fibra óptica, Wireless, Otros, Total]
Index: []

In [352]:
df_acceso_por_tecnologia['Trimestre'].unique()
#podemos observar que hay 3 valores que tienen un caracter que hay que quitar

array([4, 3, 2, 1, '3 *', '2 *', '1 *'], dtype=object)

In [353]:
df_acceso_por_tecnologia.loc[df_acceso_por_tecnologia['Trimestre']=='1 *','Trimestre']=1
df_acceso_por_tecnologia.loc[df_acceso_por_tecnologia['Trimestre']=='2 *','Trimestre']=2
df_acceso_por_tecnologia.loc[df_acceso_por_tecnologia['Trimestre']=='3 *','Trimestre']=3

In [354]:
df_acceso_por_tecnologia['Trimestre'].unique()

array([4, 3, 2, 1], dtype=object)

In [355]:
df_acceso_por_tecnologia['Provincia'].unique()

array(['Buenos Aires', 'Capital Federal', 'Catamarca', 'Chaco', 'Chubut',
       'Córdoba', 'Corrientes', 'Entre Ríos', 'Formosa', 'Jujuy',
       'La Pampa', 'La Rioja', 'Mendoza', 'Misiones', 'Neuquén',
       'Río Negro', 'Salta', 'San Juan', 'San Luis', 'Santa Cruz',
       'Santa Fe', 'Santiago Del Estero', 'Tierra Del Fuego', 'Tucumán'],
      dtype=object)

Hay 24 provincias

In [356]:
df_acceso_por_tecnologia.isna().sum()

Año             0
Trimestre       0
Provincia       0
ADSL            0
Cablemodem      0
Fibra óptica    0
Wireless        0
Otros           0
Total           0
dtype: int64

In [357]:
df_acceso_por_tecnologia.duplicated().sum()

np.int64(0)

In [358]:
#Modificamos el tipo de dato de año y trimestre
df_acceso_por_tecnologia['Año'] = df_acceso_por_tecnologia['Año'].astype('int64')
df_acceso_por_tecnologia['Trimestre'] = df_acceso_por_tecnologia['Trimestre'].astype('int64')

In [359]:
# Crear la conexión a la base de datos
engine = crear_conexion_mysql()

In [360]:
#Crear un objeto MetaData
metadata = MetaData()

# Definir la estructura de la tabla
tabla = Table(
    'internet_accesos_tecnologia', metadata,
    Column('idTabla', Integer, primary_key=True,autoincrement="auto" ),
    Column('año', Integer, nullable=False),
    Column('trimestre', Integer,nullable=False),
    Column('provincia', String(50)),
    Column('adsl', Float),
    Column('cableModem', Float),
    Column('fibraOptica', Float),
    Column('wireless', Float),
    Column('otros', Float),
    Column('total', Float)
)

In [361]:
metadata.create_all(engine)

In [364]:
df_acceso_por_tecnologia.columns

Index(['año', 'trimestre', 'provincia', 'adsl', 'cableModem', 'fibraOptica',
       'wireless', 'otros', 'total'],
      dtype='object')

In [363]:
df_acceso_por_tecnologia=df_acceso_por_tecnologia.rename(columns={
    'Año':'año' ,
    'Trimestre':'trimestre' ,
    'Provincia':'provincia' ,
    'ADSL':'adsl' ,
    'Cablemodem':'cableModem' ,
    'Fibra óptica':'fibraOptica' ,
    'Wireless':'wireless' ,
    'Otros':'otros' ,
    'Total':'total' 
})

In [365]:
# Nombre de la tabla en la base de datos
tablename = 'internet_accesos_tecnologia'

# Subir el DataFrame a la base de datos
df_acceso_por_tecnologia.to_sql(tablename, con=engine, if_exists='append', index=False)



960

# 2.- Accesos por velocidad

In [10]:
df_acceso_por_velocidad=pd.read_excel('../DatasetCrudos/Internet.xlsx',sheet_name='Accesos por velocidad') 

In [11]:
df_acceso_por_velocidad.describe

<bound method NDFrame.describe of       Año  Trimestre            Provincia  HASTA 512 kbps  \
0    2024          4         Buenos Aires         26909.0   
1    2024          4      Capital Federal           516.0   
2    2024          4            Catamarca            71.0   
3    2024          4                Chaco           236.0   
4    2024          4               Chubut           111.0   
..    ...        ...                  ...             ...   
955  2014          1           Santa Cruz           161.0   
956  2014          1             Santa Fe          8456.0   
957  2014          1  Santiago Del Estero          1234.0   
958  2014          1     Tierra Del Fuego            12.0   
959  2014          1              Tucumán             6.0   

     + 512 Kbps - 1 Mbps  + 1 Mbps - 6 Mbps  + 6 Mbps - 10 Mbps  \
0                23962.0           228393.0            239339.0   
1                 4544.0            26409.0             48083.0   
2                  209.0        

In [12]:
df_acceso_por_velocidad.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 960 entries, 0 to 959
Data columns (total 12 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   Año                  960 non-null    int64  
 1   Trimestre            960 non-null    int64  
 2   Provincia            960 non-null    object 
 3   HASTA 512 kbps       960 non-null    float64
 4   + 512 Kbps - 1 Mbps  960 non-null    float64
 5   + 1 Mbps - 6 Mbps    960 non-null    float64
 6   + 6 Mbps - 10 Mbps   960 non-null    float64
 7   + 10 Mbps - 20 Mbps  960 non-null    float64
 8   + 20 Mbps - 30 Mbps  960 non-null    float64
 9   + 30 Mbps            960 non-null    float64
 10  OTROS                954 non-null    float64
 11  Total                960 non-null    float64
dtypes: float64(9), int64(2), object(1)
memory usage: 90.1+ KB


In [13]:
df_acceso_por_velocidad.isna().sum()

Año                    0
Trimestre              0
Provincia              0
HASTA 512 kbps         0
+ 512 Kbps - 1 Mbps    0
+ 1 Mbps - 6 Mbps      0
+ 6 Mbps - 10 Mbps     0
+ 10 Mbps - 20 Mbps    0
+ 20 Mbps - 30 Mbps    0
+ 30 Mbps              0
OTROS                  6
Total                  0
dtype: int64

In [22]:
df_acceso_por_velocidad.duplicated().sum()

np.int64(0)

In [367]:
df_acceso_por_velocidad['OTROS']=df_acceso_por_velocidad['OTROS'].fillna(0)

In [369]:
# Crear la conexión a la base de datos
engine = crear_conexion_mysql()

In [370]:
#Crear un objeto MetaData
metadata = MetaData()

# Definir la estructura de la tabla df_acceso_por_velocidad
tabla = Table(
    'internet_accesos_velocidad', metadata,
    Column('idTabla', Integer, primary_key=True,autoincrement="auto" ),
    Column('año', Integer, nullable=False),
    Column('trimestre', Integer,nullable=False),
    Column('provincia', String(50)),
    Column('hasta512', Float),
    Column('De512_1mbps', Float),
    Column('De1_6mbps', Float),
    Column('De6_10mbps', Float),
    Column('De10_20mbps', Float),
    Column('De20_30mbps', Float),
    Column('mas30mbps', Float),
    Column('otros', Float),
    Column('total', Float)
)

In [371]:
metadata.create_all(engine)

In [374]:
df_acceso_por_velocidad.columns

Index(['año', 'trimestre', 'provincia', 'hasta512', 'De512_1mbps', 'De1_6mbps',
       'De6_10mbps', 'De10_20mbps', 'De20_30mbps', 'mas30mbps', 'otros',
       'total'],
      dtype='object')

In [373]:
df_acceso_por_velocidad=df_acceso_por_velocidad.rename(columns={
    'Año':'año' ,
    'Trimestre':'trimestre' ,
    'Provincia':'provincia' ,
    'HASTA 512 kbps':'hasta512' ,
    '+ 512 Kbps - 1 Mbps':'De512_1mbps' ,
    '+ 1 Mbps - 6 Mbps':'De1_6mbps' ,
    '+ 6 Mbps - 10 Mbps':'De6_10mbps' ,
    '+ 10 Mbps - 20 Mbps':'De10_20mbps' ,
    '+ 20 Mbps - 30 Mbps':'De20_30mbps',
    '+ 30 Mbps':'mas30mbps',
    'OTROS':'otros',
    'Total':'total', 
})

In [375]:
# Nombre de la tabla en la base de datos
tablename = 'internet_accesos_velocidad'

# Subir el DataFrame a la base de datos
df_acceso_por_velocidad.to_sql(tablename, con=engine, if_exists='append', index=False)

960

# 3.- Penetración en la población

In [14]:
df_penetracion_poblacion=pd.read_excel('../DatasetCrudos/Internet.xlsx',sheet_name='Penetración-poblacion') 

In [15]:
df_penetracion_poblacion.describe()

Año   Trimestre  Accesos por cada 100 hab
count   960.000000  960.000000                960.000000
mean   2018.500000    2.500000                 15.394079
std       2.873778    1.118617                  9.168959
min    2014.000000    1.000000                  2.684669
25%    2016.000000    1.750000                  8.764189
50%    2018.500000    2.500000                 13.417453
75%    2021.000000    3.250000                 19.657928
max    2023.000000    4.000000                 52.236739

In [16]:
df_penetracion_poblacion.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 960 entries, 0 to 959
Data columns (total 4 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Año                       960 non-null    int64  
 1   Trimestre                 960 non-null    int64  
 2   Provincia                 960 non-null    object 
 3   Accesos por cada 100 hab  960 non-null    float64
dtypes: float64(1), int64(2), object(1)
memory usage: 30.1+ KB


In [17]:
df_penetracion_poblacion.isna().sum()


Año                         0
Trimestre                   0
Provincia                   0
Accesos por cada 100 hab    0
dtype: int64

In [18]:
df_penetracion_poblacion.duplicated().sum()

np.int64(0)

In [23]:
# Crear la conexión a la base de datos
engine = crear_conexion_mysql()

In [24]:
df_penetracion_poblacion.columns

Index(['Año', 'Trimestre', 'Provincia', 'Accesos por cada 100 hab'], dtype='object')

In [26]:
#Crear un objeto MetaData
metadata = MetaData()

# Definir la estructura de la tabla df_acceso_por_velocidad
tabla = Table(
    'internet_penetracion_poblacion', metadata,
    Column('idTabla', Integer, primary_key=True,autoincrement="auto" ),
    Column('año', Integer, nullable=False),
    Column('trimestre', Integer,nullable=False),
    Column('provincia', String(50)),
    Column('accesosPor100Habitantes', Float)
)

In [27]:
metadata.create_all(engine)

In [28]:
df_penetracion_poblacion=df_penetracion_poblacion.rename(columns={
    'Año':'año' ,
    'Trimestre':'trimestre' ,
    'Provincia':'provincia' ,
    'Accesos por cada 100 hab':'accesosPor100Habitantes' 
})

In [29]:
# Nombre de la tabla en la base de datos
tablename = 'internet_penetracion_poblacion'

# Subir el DataFrame a la base de datos
df_penetracion_poblacion.to_sql(tablename, con=engine, if_exists='append', index=False)

960

# 4.- Penetración en hogares

In [38]:
df_penetracion_hogares=pd.read_excel('../DatasetCrudos/Internet.xlsx',sheet_name='Penetracion-hogares') 

In [39]:
df_penetracion_hogares.describe()

Año   Trimestre  Accesos por cada 100 hogares
count   960.000000  960.000000                    960.000000
mean   2018.500000    2.500000                     51.662405
std       2.873778    1.118617                     23.898192
min    2014.000000    1.000000                      9.347208
25%    2016.000000    1.750000                     32.856870
50%    2018.500000    2.500000                     48.817167
75%    2021.000000    3.250000                     65.116858
max    2023.000000    4.000000                    124.061394

In [40]:
df_penetracion_hogares.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 960 entries, 0 to 959
Data columns (total 4 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   Año                           960 non-null    int64  
 1   Trimestre                     960 non-null    int64  
 2   Provincia                     960 non-null    object 
 3   Accesos por cada 100 hogares  960 non-null    float64
dtypes: float64(1), int64(2), object(1)
memory usage: 30.1+ KB


In [41]:
df_penetracion_hogares.isna().sum()


Año                             0
Trimestre                       0
Provincia                       0
Accesos por cada 100 hogares    0
dtype: int64

In [42]:
df_penetracion_hogares.duplicated().sum()

np.int64(0)

In [43]:
engine = crear_conexion_mysql()


In [44]:
df_penetracion_hogares.columns

Index(['Año', 'Trimestre', 'Provincia', 'Accesos por cada 100 hogares'], dtype='object')

In [45]:
#Crear un objeto MetaData
metadata = MetaData()


In [46]:
# Definir la estructura de la tabla df_acceso_por_velocidad
tabla = Table(
    'internet_penetracion_hogares', metadata,
    Column('idTabla', Integer, primary_key=True,autoincrement="auto" ),
    Column('año', Integer, nullable=False),
    Column('trimestre', Integer,nullable=False),
    Column('provincia', String(50)),
    Column('accesosPor100Hogares', Float)
)
metadata.create_all(engine)

In [47]:
df_penetracion_hogares=df_penetracion_hogares.rename(columns={
    'Año':'año' ,
    'Trimestre':'trimestre' ,
    'Provincia':'provincia' ,
    'Accesos por cada 100 hogares':'accesosPor100Hogares' 
})

In [48]:
# Nombre de la tabla en la base de datos
tablename = 'internet_penetracion_hogares'

In [49]:
# Subir el DataFrame a la base de datos
df_penetracion_hogares.to_sql(tablename, con=engine, if_exists='append', index=False)

960

# 5.- Accesos por tecnología y localidad

In [97]:
df_accesos_localidad=pd.read_excel('../DatasetCrudos/Internet.xlsx',sheet_name='Accesos_tecnologia_localidad') 

In [98]:
df_accesos_localidad.describe()


ADSL    CABLEMODEM      DIAL UP   FIBRA OPTICA         OTROS  \
count   3076.000000  3.076000e+03  3076.000000    3076.000000   3076.000000   
mean     271.908322  1.957910e+03     3.833875    1270.540637     57.809818   
std     1550.899791  2.315212e+04    43.153830    8467.801469    793.944335   
min        0.000000  0.000000e+00     0.000000       0.000000      0.000000   
25%        0.000000  0.000000e+00     0.000000       0.000000      0.000000   
50%        0.000000  0.000000e+00     0.000000       1.000000      0.000000   
75%       78.250000  8.500000e+00     0.000000     152.000000      0.000000   
max    58700.000000  1.198835e+06  2050.000000  269623.000000  27703.000000   

         SATELITAL        WIMAX      WIRELESS  Total general  
count  3076.000000  3076.000000   3076.000000   3.076000e+03  
mean      0.972692     0.711313    190.429129   3.754116e+03  
std      13.648046    15.093030    712.357143   3.119100e+04  
min       0.000000     0.000000      0.000000   0.000000e+00  
25%       0.000000     0.000000      0.000000   2.700000e+01  
50%       0.000000     0.000000     28.000000   1.675000e+02  
75%       1.000000     0.000000    127.000000   1.048250e+03  
max     742.000000   444.000000  19906.000000   1.474367e+06

In [99]:
df_accesos_localidad.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3076 entries, 0 to 3075
Data columns (total 13 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   Provincia      3076 non-null   object
 1   Partido        3076 non-null   object
 2   Localidad      3076 non-null   object
 3   Link Indec     3076 non-null   object
 4   ADSL           3076 non-null   int64 
 5   CABLEMODEM     3076 non-null   int64 
 6   DIAL UP        3076 non-null   int64 
 7   FIBRA OPTICA   3076 non-null   int64 
 8   OTROS          3076 non-null   int64 
 9   SATELITAL      3076 non-null   int64 
 10  WIMAX          3076 non-null   int64 
 11  WIRELESS       3076 non-null   int64 
 12  Total general  3076 non-null   int64 
dtypes: int64(9), object(4)
memory usage: 312.5+ KB


In [100]:
df_accesos_localidad.isna().sum()


Provincia        0
Partido          0
Localidad        0
Link Indec       0
ADSL             0
CABLEMODEM       0
DIAL UP          0
FIBRA OPTICA     0
OTROS            0
SATELITAL        0
WIMAX            0
WIRELESS         0
Total general    0
dtype: int64

In [101]:
df_accesos_localidad.duplicated().sum()

np.int64(0)

In [102]:
df_accesos_localidad.head(3)

Provincia     Partido          Localidad Link Indec  ADSL  CABLEMODEM  \
0  BUENOS AIRES  25 de Mayo         25 de Mayo    6854100   756        4600   
1  BUENOS AIRES  25 de Mayo          Del Valle    6854020   181           0   
2  BUENOS AIRES  25 de Mayo  Gobernador Ugarte    6854040     0           0   

   DIAL UP  FIBRA OPTICA  OTROS  SATELITAL  WIMAX  WIRELESS  Total general  
0        0             3      0        742      0       726           6827  
1        0             0     10          1      0         0            192  
2        0             0      0          0      0       181            181

In [128]:
engine = crear_conexion_mysql()

In [129]:
df_accesos_localidad.columns

Index(['provincia', 'partido', 'localidad', 'linkIndec', 'adsl', 'cableModem',
       'dialUp', 'fibraOptica', 'otros', 'satelital', 'wimax', 'wireless',
       'totalGeneral'],
      dtype='object')

In [130]:
#Crear un objeto MetaData
metadata = MetaData()

In [131]:
# Definir la estructura de la tabla df_acceso_por_velocidad
tabla = Table(
    'internet_accesos_localidad', metadata,
    Column('idTabla', Integer, primary_key=True,autoincrement="auto" ),
    Column('provincia', String(50)),
    Column('partido', String(50)),
    Column('localidad', String(150)),
    Column('linkIndec', String(50)),
    Column('adsl', Float),
    Column('cableModem', Float),
    Column('dialUp', Float),
    Column('fibraOptica', Float),
    Column('otros', Float),
    Column('satelital', Float),
    Column('wimax', Float),
    Column('wireless', Float),
    Column('totalGeneral', Float),
)


In [132]:
metadata.create_all(engine)
df_accesos_localidad=df_accesos_localidad.rename(columns={
    'Provincia':'provincia' ,
    'Partido':'partido' ,
    'Localidad':'localidad' ,
    'Link Indec':'linkIndec',
    'ADSL':'adsl',
    'CABLEMODEM':'cableModem',
    'DIAL UP':'dialUp',
    'FIBRA OPTICA':'fibraOptica',
    'OTROS':'otros',
    'SATELITAL':'satelital',
    'WIMAX':'wimax',
    'WIRELESS':'wireless',
    'Total general':'totalGeneral'
})


In [133]:
# Nombre de la tabla en la base de datos
tablename = 'internet_accesos_localidad'

In [134]:
x=df_accesos_localidad['localidad'].iloc[2251] #df_accesos_localidad.iloc[2251]
len(x)

54

In [135]:
# Subir el DataFrame a la base de datos
df_accesos_localidad.to_sql(tablename, con=engine, if_exists='append', index=False)

3076

# 6.- Ingresos

In [143]:
df_ingresos=pd.read_excel('../DatasetCrudos/Internet.xlsx',sheet_name='Ingresos') 


In [144]:
df_ingresos.describe()


Año  Trimestre  Ingresos (miles de pesos)
count    40.000000  40.000000               4.000000e+01
mean   2018.750000   2.500000               3.281153e+07
std       3.642696   1.132277               3.850304e+07
min    2014.000000   1.000000               2.984054e+06
25%    2016.000000   1.750000               6.817891e+06
50%    2018.500000   2.500000               1.842255e+07
75%    2021.000000   3.250000               3.942974e+07
max    2033.000000   4.000000               1.673760e+08

In [145]:
df_ingresos.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 40 entries, 0 to 39
Data columns (total 4 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Año                        40 non-null     int64  
 1   Trimestre                  40 non-null     int64  
 2   Ingresos (miles de pesos)  40 non-null     float64
 3   Periodo                    40 non-null     object 
dtypes: float64(1), int64(2), object(1)
memory usage: 1.4+ KB


In [146]:
df_ingresos.isna().sum()


Año                          0
Trimestre                    0
Ingresos (miles de pesos)    0
Periodo                      0
dtype: int64

In [147]:
df_ingresos.duplicated().sum()

np.int64(0)

In [148]:
engine = crear_conexion_mysql()

In [149]:
df_ingresos.columns

Index(['Año', 'Trimestre', 'Ingresos (miles de pesos)', 'Periodo'], dtype='object')

In [150]:
#Crear un objeto MetaData
metadata = MetaData()

In [151]:
# Definir la estructura de la tabla df_acceso_por_velocidad
tabla = Table(
    'internet_ingresos', metadata,
    Column('idTabla', Integer, primary_key=True,autoincrement="auto" ),
    Column('año', Integer, nullable=False),
    Column('trimestre', Integer,nullable=False),
    Column('ingresos', Float),
    Column('periodo', String(20))
)


In [152]:
metadata.create_all(engine)

In [153]:
df_ingresos=df_ingresos.rename(columns={
    'Año':'año' ,
    'Trimestre':'trimestre' ,
    'Ingresos (miles de pesos)':'ingresos' ,
    'Periodo':'periodo' 
})

In [154]:
# Nombre de la tabla en la base de datos
tablename = 'internet_ingresos'


In [155]:
# Subir el DataFrame a la base de datos
df_ingresos.to_sql(tablename, con=engine, if_exists='append', index=False)

40

# 7 Totales Accesos por tecnología

In [136]:
df_totales=pd.read_excel('../DatasetCrudos/Internet.xlsx',sheet_name='Totales Accesos Por Tecnología') 

In [137]:
df_totales.describe()

Año  Trimestre          ADSL    Cablemodem  Fibra óptica  \
count    40.000000  40.000000  4.000000e+01  4.000000e+01  4.000000e+01   
mean   2018.500000   2.500000  2.791887e+06  4.359240e+06  1.081302e+06   
std       2.908872   1.132277  1.027693e+06  1.308010e+06  1.179722e+06   
min    2014.000000   1.000000  8.363900e+05  2.407330e+06  1.391870e+05   
25%    2016.000000   1.750000  2.001598e+06  3.102459e+06  1.694705e+05   
50%    2018.500000   2.500000  3.230908e+06  4.374635e+06  5.125560e+05   
75%    2021.000000   3.250000  3.716772e+06  5.687862e+06  1.495696e+06   
max    2023.000000   4.000000  3.803024e+06  6.073426e+06  3.908183e+06   

            Wireless          Otros         Total  
count      40.000000      40.000000  4.000000e+01  
mean   289710.900000  154847.925000  8.676987e+06  
std    197048.992979   82678.927051  1.655946e+06  
min     70749.000000   54300.000000  6.398398e+06  
25%     85307.250000   69048.000000  7.187732e+06  
50%    270359.000000  172437.000000  8.560895e+06  
75%    480829.750000  238176.500000  9.918698e+06  
max    585760.000000  265328.000000  1.154766e+07

In [138]:
df_totales.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 40 entries, 0 to 39
Data columns (total 9 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   Año           40 non-null     int64 
 1   Trimestre     40 non-null     int64 
 2   ADSL          40 non-null     int64 
 3   Cablemodem    40 non-null     int64 
 4   Fibra óptica  40 non-null     int64 
 5   Wireless      40 non-null     int64 
 6   Otros         40 non-null     int64 
 7   Total         40 non-null     int64 
 8   Periodo       40 non-null     object
dtypes: int64(8), object(1)
memory usage: 2.9+ KB


In [139]:
df_totales.isna().sum()

Año             0
Trimestre       0
ADSL            0
Cablemodem      0
Fibra óptica    0
Wireless        0
Otros           0
Total           0
Periodo         0
dtype: int64

In [140]:
df_totales.duplicated().sum()

np.int64(0)

In [141]:
engine = crear_conexion_mysql()

In [142]:
df_totales.columns

Index(['Año', 'Trimestre', 'ADSL', 'Cablemodem', 'Fibra óptica', 'Wireless',
       'Otros', 'Total', 'Periodo'],
      dtype='object')

In [65]:
#Crear un objeto MetaData
metadata = MetaData()

In [71]:
# Definir la estructura de la tabla df_acceso_por_velocidad
tabla = Table(
    'internet_accesos_totales', metadata,
    Column('idTabla', Integer, primary_key=True,autoincrement="auto" ),
    Column('año', Integer, nullable=False),
    Column('trimestre', Integer,nullable=False),
    Column('adsl', Float),
    Column('cableModem', Float),
    Column('fibraOptica', Float),
    Column('wireless', Float),
    Column('otros', Float),
    Column('total', Float),
    Column('periodo', String(20))
)


In [72]:
metadata.create_all(engine)

In [73]:
df_totales=df_totales.rename(columns={
    'Año':'año' ,
    'Trimestre':'trimestre' ,
    'ADSL':'adsl' ,
    'Cablemodem':'cableModem',
    'Fibra óptica':'fibraOptica',
    'Wireless':'wireless',
    'Otros':'otros',
    'Total':'total',
    'Periodo':'periodo'
})


In [74]:
# Nombre de la tabla en la base de datos
tablename = 'internet_accesos_totales'

# Subir el DataFrame a la base de datos
df_totales.to_sql(tablename, con=engine, if_exists='append', index=False)

40

# 8 Portabilidad

In [4]:
df_portabilidad=pd.read_excel('../DatasetCrudos/Portabilidad.xlsx',sheet_name='Portabilidad') 

In [5]:
df_portabilidad.describe()

Año         Mes      Personal         Claro       Nextel   \
count   139.000000  139.000000    139.000000    139.000000   139.000000   
mean   2017.460432    6.474820   1420.316547   4436.827338 -1231.007194   
min    2012.000000    1.000000 -18668.000000 -42345.000000 -4964.000000   
25%    2015.000000    4.000000  -4243.000000  -1179.500000 -1873.500000   
50%    2017.000000    6.000000   -200.000000   4447.000000 -1267.000000   
75%    2020.000000    9.000000   3962.500000  10492.500000     0.000000   
max    2023.000000   12.000000  41931.000000  38116.000000     0.000000   
std       3.367075    3.407857  10801.265898  14107.432385  1195.870113   

          Movistar         Otros                       Periodo   
count    139.000000   139.000000                            139  
mean   -4666.920863    40.784173  2017-11-30 17:36:41.438849024  
min   -63567.000000     0.000000            2012-03-01 00:00:00  
25%   -12456.000000     0.000000            2015-01-16 12:00:00  
50%    -3746.000000     0.000000            2017-12-01 00:00:00  
75%     2119.500000     0.000000            2020-10-16 12:00:00  
max    41453.000000  1050.000000            2023-09-01 00:00:00  
std    14798.534389   148.059563                            NaN

In [6]:
df_portabilidad.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 139 entries, 0 to 138
Data columns (total 8 columns):
 #   Column      Non-Null Count  Dtype         
---  ------      --------------  -----         
 0    Año        139 non-null    int64         
 1    Mes        139 non-null    int64         
 2    Personal   139 non-null    int64         
 3    Claro      139 non-null    int64         
 4    Nextel     139 non-null    int64         
 5    Movistar   139 non-null    int64         
 6   Otros       139 non-null    int64         
 7    Periodo    139 non-null    datetime64[ns]
dtypes: datetime64[ns](1), int64(7)
memory usage: 8.8 KB


In [7]:
df_portabilidad.isna().sum()

 Año          0
 Mes          0
 Personal     0
 Claro        0
 Nextel       0
 Movistar     0
Otros         0
 Periodo      0
dtype: int64

In [8]:
df_portabilidad.duplicated().sum()

np.int64(0)

In [12]:
engine = crear_conexion_mysql()

In [13]:
df_portabilidad.columns

Index([' Año ', ' Mes ', ' Personal ', ' Claro ', ' Nextel ', ' Movistar ',
       'Otros', ' Periodo '],
      dtype='object')

In [14]:
#Crear un objeto MetaData
metadata = MetaData()

In [15]:
# Definir la estructura de la tabla df_acceso_por_velocidad
tabla = Table(
    'internet_portabilidad', metadata,
    Column('idTabla', Integer, primary_key=True,autoincrement="auto" ),
    Column('año', Integer, nullable=False),
    Column('mes', Integer,nullable=False),
    Column('Personal', Integer),
    Column('claro', Integer),
    Column('nextel', Integer),
    Column('movistar', Integer),
    Column('otros', Integer),
    Column('periodo', String(20))
)

In [16]:
metadata.create_all(engine)

In [21]:
df_portabilidad=df_portabilidad.rename(columns={
    ' Año ':'año' ,
    ' Mes ':'mes' ,
    ' Personal ':'personal' ,
    ' Claro ':'claro',
    ' Nextel ':'nextel',
    ' Movistar ':'movistar',
    ' Otros ':'otros',
    ' Periodo ':'periodo'
})

In [22]:
# Nombre de la tabla en la base de datos
tablename = 'internet_portabilidad'

# Subir el DataFrame a la base de datos
df_portabilidad.to_sql(tablename, con=engine, if_exists='append', index=False)

139

# 9 Portabilidad in

In [55]:
df_portabilidad_in=pd.read_excel('../DatasetCrudos/Portabilidad.xlsx',sheet_name='Portin') 

In [56]:
df_portabilidad_in.describe()

Año         Mes       Personal          Claro      Nextel  \
count   139.000000  139.000000     139.000000     139.000000   78.000000   
mean   2017.460432    6.474820   48347.568345   58076.014388  126.512821   
std       3.367075    3.407857   31554.879225   33743.575848  142.875638   
min    2012.000000    1.000000      40.000000      72.000000    0.000000   
25%    2015.000000    4.000000   27928.500000   34467.500000   12.250000   
50%    2017.000000    6.000000   43106.000000   56791.000000   91.500000   
75%    2020.000000    9.000000   63493.000000   78123.500000  174.250000   
max    2023.000000   12.000000  179629.000000  132676.000000  676.000000   

            Movistar        Otros  Total general  
count     139.000000    19.000000     139.000000  
mean    53113.985612   303.473684  159650.043165  
std     39348.102580   299.043358   98295.871674  
min        41.000000     9.000000     153.000000  
25%     13514.500000    51.500000   89669.500000  
50%     45207.000000   238.000000  139781.000000  
75%     87365.000000   430.500000  245761.500000  
max    143327.000000  1059.000000  377682.000000

In [57]:
df_portabilidad_in.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 139 entries, 0 to 138
Data columns (total 8 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Año            139 non-null    int64  
 1   Mes            139 non-null    int64  
 2   Personal       139 non-null    int64  
 3   Claro          139 non-null    int64  
 4   Nextel         78 non-null     float64
 5   Movistar       139 non-null    int64  
 6   Otros          19 non-null     float64
 7   Total general  139 non-null    int64  
dtypes: float64(2), int64(6)
memory usage: 8.8 KB


In [58]:
df_portabilidad_in.isna().sum()

Año                0
Mes                0
Personal           0
Claro              0
Nextel            61
Movistar           0
Otros            120
Total general      0
dtype: int64

In [59]:
df_portabilidad_in.duplicated().sum()

np.int64(0)

In [60]:
engine = crear_conexion_mysql()

In [61]:
df_portabilidad_in.columns

Index(['Año', 'Mes', 'Personal', 'Claro', 'Nextel', 'Movistar', 'Otros',
       'Total general'],
      dtype='object')

In [62]:
#Crear un objeto MetaData
metadata = MetaData()

In [63]:
# Definir la estructura de la tabla df_acceso_por_velocidad
tabla = Table(
    'internet_portabilidad_in', metadata,
    Column('idTabla', Integer, primary_key=True,autoincrement="auto" ),
    Column('año', Integer, nullable=False),
    Column('mes', Integer,nullable=False),
    Column('Personal', Integer),
    Column('claro', Integer),
    Column('nextel', Float),
    Column('movistar', Integer),
    Column('otros', Float),
    Column('total_general',Integer)
)

In [64]:
metadata.create_all(engine)

In [65]:
df_portabilidad_in=df_portabilidad_in.rename(columns={
    'Año':'año' ,
    'Mes':'mes' ,
    'Personal':'personal' ,
    'Claro':'claro',
    'Nextel':'nextel',
    'Movistar':'movistar',
    'Otros':'otros',
    'Total general':'total_general'
})

       

In [66]:
# Nombre de la tabla en la base de datos
tablename = 'internet_portabilidad_in'
# Subir el DataFrame a la base de datos
df_portabilidad_in.to_sql(tablename, con=engine, if_exists='append', index=False)


139

# Mapa conectividad

In [20]:
df_conectividad=pd.read_excel('../DatasetCrudos/mapa_conectividad.xlsx',sheet_name='Hoja3') 

In [21]:
df_conectividad.describe()

Población          Link      Latitud     Longitud
count  4.311000e+03  4.299000e+03  4311.000000  4311.000000
mean   8.695088e+03  3.890626e+07   -32.261794   -63.039123
std    3.821965e+04  2.920333e+07     4.999418     3.830260
min    0.000000e+00  2.000010e+06   -77.874009   -72.891627
25%    1.160000e+02  1.007702e+07   -34.707626   -65.687041
50%    5.760000e+02  3.403505e+07   -32.283511   -63.450636
75%    3.203000e+03  7.005610e+07   -28.584663   -60.008572
max    1.317298e+06  9.401402e+07   -21.817293   -34.626560

In [22]:
df_conectividad.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4311 entries, 0 to 4310
Data columns (total 16 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Provincia       4311 non-null   object 
 1   Partido         4311 non-null   object 
 2   Localidad       4311 non-null   object 
 3   Población       4311 non-null   int64  
 4   ADSL            4311 non-null   object 
 5   Cablemódem      4311 non-null   object 
 6   Dial Up         4311 non-null   object 
 7   Fibra óptica    4311 non-null   object 
 8   Satelital       4311 non-null   object 
 9   Wireless        4311 non-null   object 
 10  Telefonía Fija  4311 non-null   object 
 11  3G              4311 non-null   object 
 12  4G              4311 non-null   object 
 13  Link            4299 non-null   float64
 14  Latitud         4311 non-null   float64
 15  Longitud        4311 non-null   float64
dtypes: float64(3), int64(1), object(12)
memory usage: 539.0+ KB


In [23]:
df_conectividad.isna().sum()

Provincia          0
Partido            0
Localidad          0
Población          0
ADSL               0
Cablemódem         0
Dial Up            0
Fibra óptica       0
Satelital          0
Wireless           0
Telefonía Fija     0
3G                 0
4G                 0
Link              12
Latitud            0
Longitud           0
dtype: int64

In [24]:
df_conectividad.fillna(0)

Provincia       Partido                                Localidad  \
0     BUENOS AIRES    9 de Julio  Alfredo Demarchi (Est. Facundo Quiroga)   
1     BUENOS AIRES    9 de Julio                        Carlos María Naón   
2     BUENOS AIRES    9 de Julio                                 Dudignac   
3     BUENOS AIRES    9 de Julio                 La Aurora (Est. La Niña)   
4     BUENOS AIRES    9 de Julio           Manuel B. Gonnet (Est. French)   
...            ...           ...                                      ...   
4306      SANTA FE   Castellanos                                 Virginia   
4307      SANTA FE   Castellanos                            Zenón Pereyra   
4308      SANTA FE  Constitución                                  Alcorta   
4309      SANTA FE  Constitución                  Barrio Arroyo del Medio   
4310      SANTA FE  Constitución                             Barrio Mitre   

      Población ADSL Cablemódem Dial Up Fibra óptica Satelital Wireless  \
0          1853   Sí         --      --           --        --       --   
1           497   --         --      --           --        --       --   
2          2670   --         --      Sí           --        --       Sí   
3           531   --         --      --           --        --       --   
4           748   Sí         --      --           Sí        --       --   
...         ...  ...        ...     ...          ...       ...      ...   
4306        151   --         --      --           --        --       Sí   
4307       1637   --         Sí      --           --        --       --   
4308       7450   Sí         Sí      --           Sí        --       Sí   
4309        310   --         --      --           --        --       --   
4310        816   --         --      --           --        --       --   

     Telefonía Fija  3G  4G       Link    Latitud   Longitud  
0                Sí  Sí  Sí  6588010.0 -35.292982 -61.407254  
1                Sí  Sí  Sí  6588020.0 -35.239499 -60.825136  
2                Sí  Sí  Sí  6588040.0 -35.651193 -60.709896  
3                --  --  Sí  6588050.0 -35.407947 -61.210567  
4                Sí  Sí  Sí  6588060.0 -35.520141 -60.998949  
...             ...  ..  ..        ...        ...        ...  
4306             --  --  --  6638040.0 -30.740253 -61.340967  
4307             --  Sí  Sí  6638040.0 -31.564397 -61.898425  
4308             Sí  Sí  Sí  6638040.0 -33.540198 -61.124634  
4309             --  --  --  6638040.0 -33.286067 -60.272098  
4310             --  Sí  Sí  6638040.0 -33.255595 -60.389776  

[4311 rows x 16 columns]

In [25]:
engine = crear_conexion_mysql()

In [26]:
df_conectividad.columns

Index(['Provincia', 'Partido', 'Localidad', 'Población', 'ADSL', 'Cablemódem',
       'Dial Up', 'Fibra óptica', 'Satelital', 'Wireless', 'Telefonía Fija',
       '3G', '4G', 'Link', 'Latitud', 'Longitud'],
      dtype='object')

In [27]:
#Crear un objeto MetaData
metadata = MetaData()

In [28]:
# Definir la estructura de la tabla df_acceso_por_velocidad
tabla = Table(
    'internet_conectividad', metadata,
    Column('idTabla', Integer, primary_key=True,autoincrement="auto" ),
    Column('Provincia', String(80)),
    Column('Partido', String(80)),
    Column('Localidad', String(80)),
    Column('Población', Integer),
    Column('ADSL', String(10)),
    Column('Cablemódem', String(10)),
    Column('Dial Up', String(10)),
    Column('Fibra Óptica', String(10)),
    Column('Satelital',String(10)),
    Column('Wireless',String(10)),
    Column('Telefonía Fija',String(10)),
    Column('3G',String(10)),
    Column('4G',String(10)),
    Column('Link',Float),
    Column('Latitud',Float),
    Column('Longitud',Float)
)

In [29]:
metadata.create_all(engine)

In [30]:
# Nombre de la tabla en la base de datos
tablename = 'internet_conectividad'
# Subir el DataFrame a la base de datos
df_conectividad.to_sql(tablename, con=engine, if_exists='append', index=False)

4311

# 11 Empresas por provincia

Extracción de datos de un archivo externo en formato pdf

In [1]:
import pdfplumber
import pandas as pd


In [3]:
# Especifica la ruta del archivo PDF
pdf_path = "../DatasetCrudos/isp_por_provincias_2.pdf"

In [5]:
dfs = []
# Función para verificar si una tabla tiene una estructura válida
def is_valid_table(table):
    if not table:
        return False
    if len(table) < 2:
        return False
    if any(len(row) != len(table[0]) for row in table):
        return False
    return True

# Abre el archivo PDF con pdfplumber
with pdfplumber.open(pdf_path) as pdf:
    pages = pdf.pages[1:-2]
    # Iterar a través de cada página
    for page in pages:
        # Extrae las tablas de la página
        tables = page.extract_tables()
        
        # Depuración: Imprimir las tablas extraídas
        for i, table in enumerate(tables):
            print(f"Tabla {i+1} en la página {page.page_number}: {table}")

        # Convierte cada tabla en un DataFrame de pandas y añádela a la lista si es válida
        for table in tables:
            if is_valid_table(table):
                df = pd.DataFrame(table[1:], columns=table[0])
                #df.columns = df.columns.str.strip()  # Eliminar espacios en blanco de los nombres de las columnas
                print(df.shape)
                dfs.append(df.reset_index(drop=True))# Resetea los índices 

Tabla 1 en la página 2: [['Miembro Telefono Empresa Web Localidad Provincia', None, None, None, None], ['GLOBALSAT ARGENTINA TELECOMUNICACIONES SRL', '54 11 3987 7728 www.globalsatgroup.com.ar', None, '', 'ARGENTINA'], ['INTELIGLOBE COMUNICACIONES ARGENTINA SRL (IG', '+N5E4T W11O 5R3K6S7) 1940', 'www.ignetworks.com.ar', '', 'ARGENTINA'], ['NUAGE SYSTEMS S.A (EMBLUE)', '1141363500', 'www.embluemail.com', '', 'ARGENTINA'], ['SERVICIO SATELITAL S.A.', '+54 11 4317 1800', 'www.satelital.com.ar', '', 'ARGENTINA'], ['TELESPAZIO ARGENTINA SA', '(011) 4852 8700', 'www.telespazio.com.ar', '', 'ARGENTINA'], ['TESAM ARGENTINA S.A.', '(+54) (11) 4114 2200 - 0810-34', 'www.tesacom.net', '', 'ARGENTINA'], ['WEBHOSTING S.R.L. (WNPOWER)', '0810-345-9676', 'www.wnpower.com', '', 'ARGENTINA'], ['NODOCOOP FEDERACION DE COOPERATIVAS LTDA', '(02923) 494200', 'www.nodocoop.com.ar', '17 de Agosto', 'BUENOS AIRES'], ['INFOSPORT SA ( TELWINET)', '0220-4924-393', 'www.telwinet.com.ar', '20 de Junio', 'BUENOS AI

In [6]:
column_names = ['empresa', 'telefono', 'web', 'localidad', 'provincia']

dfs_adjusted = []

# Procesar cada DataFrame
for i, data in enumerate(dfs):
    print(f"Procesando tabla {i+1}...")

    if data.empty:
        print("Tabla vacía.")
        continue

    # Depuración: Imprimir la primera fila de datos
    print("Primera fila:", data.iloc[0].tolist())

    if data.iloc[0].tolist() == column_names:
        # Si el primer registro es el encabezado, eliminar el primer registro
        print("Encabezado detectado.")
        data = data.iloc[1:].reset_index(drop=True)
    
    # Asignar los nombres de las columnas
    data.columns = column_names
    dfs_adjusted.append(data)

# Concatenar todos los DataFrames ajustados en uno solo
final_df = pd.concat(dfs_adjusted, ignore_index=True)

Procesando tabla 1...
Primera fila: ['GLOBALSAT ARGENTINA TELECOMUNICACIONES SRL', '54 11 3987 7728 www.globalsatgroup.com.ar', None, '', 'ARGENTINA']
Procesando tabla 2...
Primera fila: ['CATTANEO LUIS E. (BBL-Net / Velostar )', '0291 452-4444', 'www.bbl-net.com.ar', 'Algarrobo', 'BUENOS AIRES']
Procesando tabla 3...
Primera fila: ['NODO NET S.A.', '', '', 'Bahia Blanca', 'BUENOS AIRES']
Procesando tabla 4...
Primera fila: ['ADVANTUN SRL', '5365-8800', 'www.advantun.com', 'Bella Vista', 'BUENOS AIRES']
Procesando tabla 5...
Primera fila: ['IFX NETWORKS ARGENTINA SRL', '(011) 5031-2400', 'www.ifxnetworks.com', 'Bernal', 'BUENOS AIRES']
Procesando tabla 6...
Primera fila: ['SION SA', '0810 999 0100', 'www.sion.com', 'Canning', 'BUENOS AIRES']
Procesando tabla 7...
Primera fila: ['IPNEXT S.A.', '5368-4000', 'www.ipnext.net.ar', 'Chacabuco', 'BUENOS AIRES']
Procesando tabla 8...
Primera fila: ['FACEBOOK ARGENTINA S.R.L.', '541150773400', 'www.facebook.com', 'Ciudad Autónoma de Buenos Air'

In [36]:
engine = crear_conexion_mysql()

In [37]:
#Crear un objeto MetaData
metadata = MetaData()

In [38]:
final_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2700 entries, 0 to 2699
Data columns (total 5 columns):
 #   Column     Non-Null Count  Dtype 
---  ------     --------------  ----- 
 0   empresa    2700 non-null   object
 1   telefono   2627 non-null   object
 2   web        2692 non-null   object
 3   localidad  2700 non-null   object
 4   provincia  2700 non-null   object
dtypes: object(5)
memory usage: 105.6+ KB


In [39]:
# Definir la estructura de la tabla df_acceso_por_velocidad
tabla = Table(
    'internet_empresas', metadata,
    Column('idTabla', Integer, primary_key=True,autoincrement="auto" ),
    Column('empresa', String(150)),
    Column('telefono', String(100)),
    Column('web', String(80)),
    Column('localidad', String(80)),
    Column('provincia', String(50))
)


In [40]:
metadata.create_all(engine)

In [41]:
# Nombre de la tabla en la base de datos
tablename = 'internet_empresas'
# Subir el DataFrame a la base de datos
final_df.to_sql(tablename, con=engine, if_exists='append', index=False)

2700

# Funciones

In [42]:
def crear_conexion_mysql():
    load_dotenv('conexion.env')
    # Leer variables de entorno
    username = os.getenv('DB_USERNAME')
    password = os.getenv('DB_PASSWORD')
    hostname = os.getenv('DB_HOST')
    dbname = os.getenv('DB_NAME')
    # Crear la conexión a la base de datos
    engine = create_engine(f'mysql+mysqlconnector://{username}:{password}@{hostname}/{dbname}')
    return engine